File: io-hotmoka-network/src/main/java/io/hotmoka/network/internal/models/transactions/TransactionReferenceModel.java
package io.hotmoka.network.internal.models.transactions;

import io.hotmoka.beans.InternalFailureException;
import io.hotmoka.beans.references.LocalTransactionReference;
import io.hotmoka.beans.references.TransactionReference;

public class TransactionReferenceModel {
    private String hash;

    public TransactionReferenceModel() {}

    public TransactionReferenceModel(TransactionReference reference) {
    	if (reference instanceof LocalTransactionReference)
    		this.hash = reference.getHash();
    	else
    		throw new InternalFailureException("unexpected transaction reference of type " + reference.getClass().getName());
    }

    public String getHash() {
        return hash;
    }

    public void setHash(String hash) {
        this.hash = hash;
    }
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/models/values/TransactionReferenceModel.kt
package io.hotmoka.network.thin.client.models.values

/**
 * The model of a transaction reference.
 */
class TransactionReferenceModel(
        /**
         * The type of transaction.
         */
        val type: String,
        /**
         * Used at least for local transactions.
         */
        val hash: String
)